pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -999,17 +985,20 @@ func validateVSphere(m *Machine, clusterID string) (bool, []string, utilerrors.A
 		errs = append(errs, field.Required(field.NewPath("providerSpec", "template"), "template must be provided"))
 	}
 
-	errs = append(errs, validateVSphereWorkspace(providerSpec.Workspace, field.NewPath("providerSpec", "workspace"))...)
+	workspaceWarnings, workspaceErrors := validateVSphereWorkspace(providerSpec.Workspace, field.NewPath("providerSpec", "workspace"))
+	warnings = append(warnings, workspaceWarnings...)
+	errs = append(errs, workspaceErrors...)
+	
 	errs = append(errs, validateVSphereNetwork(providerSpec.Network, field.NewPath("providerSpec", "network"))...)
 
 	if providerSpec.NumCPUs != 0 && providerSpec.NumCPUs < minVSphereCPU {
```

Comment:
shouldn't we remove the `if providerSpec.NumCPUs != 0` check? so wouldn't `providerSpec.NumCPUs < minVSphereCPU` be enough?

Reply:
Yep, good point, will fix